viewing_party/party.py

```diff
@@ -1,23 +1,177 @@
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            return user_data
+    return user_data
 
-# -----------------------------------------
 # ------------- WAVE 2 --------------------
-# -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    sum = 0
+    average = 0
+
+    for i in range(len(user_data["watched"])):
+        sum += user_data["watched"][i]["rating"]
+        average = sum  / len(user_data["watched"])
+
+    return average
+
+def get_most_watched_genre(user_data):
+
+    index = 0
+    genres = []
+    least_popular = []
+    most_popular = []
+    
+    if user_data == {"watched":[]}:
+        return None
+    else:
+        for movie in user_data["watched"]:
+            index +=1
+            genres.append(user_data["watched"][index-1]["genre"])
+        for item in genres:
+            if item not in least_popular:
+                least_popular.append(item)
+            else:
+                most_popular.append(item)
+        return (most_popular[0])
 
-# -----------------------------------------
 # ------------- WAVE 3 --------------------
-# -----------------------------------------
 
-        
-# -----------------------------------------
-# ------------- WAVE 4 --------------------
-# -----------------------------------------
+def get_unique_watched(user_data):
+    unique = []
+    not_unique = []
+    total_movies_list = []
+    users_movies = []
+    friends_movies = []
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+    total_movies_list = users_movies + friends_movies
+    
+    for item in total_movies_list:
+        if item not in friends_movies:
+            unique.append(item)
+        else:
+            not_unique.append(item)
+    return unique
+
+def get_friends_unique_watched(user_data):
+    unique = []
+    users_movies = []
+    friends_movies = []
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+            
+    for item in friends_movies:
+        if item not in users_movies:
+            if item not in unique:
+                unique.append(item)
+    
+    return unique
+# 
+# ------------- WAVE 4 -------------
+
+def get_available_recs(user_data):
+    unique = []
+    users_movies = []
+    friends_movies = []
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+            
+    for item in friends_movies:
+        if item not in users_movies:
+            if item not in unique:

```

Comment:
This would have been a great place to reuse your `get_friends_unique_watched` function! An example of how this would look:
```
unique_friend_movies = get_friends_unique_watched(user_data)
recs = []

    for movie in unique_friend_movies:
        if movie['host'] in user_data['subscriptions']:
            recs.append(movie)

return recs
```